tests/components/forecast_solar/test_config_flow.py

```diff
@@ -58,66 +58,63 @@ async def test_user_flow(hass: HomeAssistant) -> None:
 
 
 async def test_options_flow_invalid_api(
-    hass: HomeAssistant, mock_config_entry: MockConfigEntry
+    hass: HomeAssistant, init_integration: MockConfigEntry
 ) -> None:
     """Test options config flow when API key is invalid."""
-    mock_config_entry.add_to_hass(hass)
-    with patch(
-        "homeassistant.components.forecast_solar.async_setup_entry", return_value=True
-    ):
-        await hass.config_entries.async_setup(mock_config_entry.entry_id)
```

Comment:
We need to set up the integration before starting the options flow to make sure the config flow module is loaded.

Reply:
That is set up by the `init_integration` fixture

Reply:
Good. We still need to patch the integration setup function when doing this though.

Reply:
Do you have an example of an integration that does this? Because both Frenck and I can't really figure out why this is still necessary.

Reply:
> Good. We still need to patch the integration setup function when doing this though.

Why? The client library is fully patched (the same fixture used by all other tests). There is no direct need to patch any HA internals in this case.

I do think the whole lead-up to these comments is indeed a bit odd, as per Klaas above. I could not find any integration that actually tests the reloading functionality of a config entry when the options change in an options flow test.

It might not be bad to do it, but it is not something we normally do or test on the integration level.

Reply:
We normally patch the integration setup function in config flow tests to avoid setting up the whole integration, when that is a possibility.

Reply:
Actually, we don't. One of the few cases I found that actually tests this, is the unifiprotect integration, which actually does set it up fully (and actually caused tests errors today because of it).

Do we have other integrations that are handling this?

Reply:
Adjusted, patched out config entry setup.